quartz/util/encryption.ts

```diff
@@ -0,0 +1,527 @@
+    )
+  }
+
+  // Build result object
+  const result: EncryptionResult = {
+    encryptedContent: arrayBufferToHex(encryptedBuffer),
+    encryptionSalt: encryptionHashData.salt,
+    iv: arrayBufferToHex(initializationVector.buffer as ArrayBuffer),
+  }
+
+  if (authTag) {
+    result.authTag = arrayBufferToHex(authTag)
+  }
+
+  return result
+}
+
+export async function decryptContent(
+  encrypted: EncryptionResult,
+  config: CompleteCryptoConfig,
+  password: string,
+): Promise<string> {
+  checkCryptoSupport()
+  const { encryptedContent, encryptionSalt, iv, authTag } = encrypted
+
+  // Create encryption hash and derive key
+  const encryptionSaltBuffer = hexToArrayBuffer(encryptionSalt)
+  const encryptionHashData = await hashString(password, encryptionSaltBuffer)
+  const key = await deriveKeyFromHash(encryptionHashData.hash, config.algorithm)
+
+  // Prepare for decryption
+  const ciphertext = hexToArrayBuffer(encryptedContent)
+  let decryptedBuffer: ArrayBuffer
+
+  try {
+    if (config.algorithm === "aes-256-gcm") {
+      // GCM mode
+      if (!iv) throw new Error("IV is required for GCM mode")
+      if (!authTag) throw new Error("Authentication tag is required for GCM mode")
+
+      const initializationVectorBuffer = hexToArrayBuffer(iv)
+      const authTagBuffer = hexToArrayBuffer(authTag)
+
+      // For GCM decryption, we need to append the auth tag to the ciphertext
+      const combined = new Uint8Array(ciphertext.byteLength + authTagBuffer.byteLength)
+      combined.set(new Uint8Array(ciphertext), 0)
+      combined.set(new Uint8Array(authTagBuffer), ciphertext.byteLength)
+
+      decryptedBuffer = await crypto.subtle.decrypt(
+        {
+          name: "AES-GCM",
+          iv: initializationVectorBuffer,
+        },
+        key,
+        combined.buffer,
+      )
+    } else if (config.algorithm === "aes-256-cbc") {
+      // CBC mode
+      if (!iv) throw new Error("IV is required for CBC mode")
+      const initializationVectorBuffer = hexToArrayBuffer(iv)
+
+      decryptedBuffer = await crypto.subtle.decrypt(
+        {
+          name: "AES-CBC",
+          iv: initializationVectorBuffer,
+        },
+        key,
+        ciphertext,
+      )
+    } else {
+      throw new Error("Unsupported algorithm: " + config.algorithm)
+    }
+  } catch (error) {
+    throw new Error(
+      `Decryption failed: ${error instanceof Error ? error.message : "Unknown error"}`,
+    )
+  }
+
+  return arrayBufferToString(decryptedBuffer)
+}
+
+export async function searchForValidPassword(
+  filePath: string,
+  hash: Hash,
+  config: CompleteCryptoConfig,
+  blacklist: Set<string> | null = null,
+): Promise<string | undefined> {
+  const passwords = getRelevantPasswords(filePath)
+
+  for (const password of passwords) {
+    if (blacklist && blacklist.has(password)) {
+      continue
+    }
+
+    if (await verifyPasswordHash(password, hash)) {
+      addPasswordToCache(password, filePath, config.ttl)
```

Comment:
let's just not do any cache here, just more susceptible to CVE and makes it more complicated.

we can just enforce typing the password again.